packages/remix-forms/src/createForm.tsx

```diff
@@ -333,34 +345,45 @@ function createForm({
       } as Field<SchemaType>
     }
 
-    const hiddenFieldsErrorsToGlobal = (globalErrors: string[] = []) => {
-      const deepHiddenFieldsErrors = hiddenFields?.map((hiddenField) => {
-        const hiddenFieldErrors = fieldErrors(hiddenField)
-
-        if (hiddenFieldErrors instanceof Array) {
-          const hiddenFieldLabel =
-            (labels && labels[hiddenField]) || inferLabel(String(hiddenField))
-          return hiddenFieldErrors.map(
-            (error) => `${hiddenFieldLabel}: ${error}`,
-          )
-        } else return []
-      })
-      const hiddenFieldsErrors: string[] = deepHiddenFieldsErrors?.flat() || []
+    const hiddenFieldsErrorsToGlobal = React.useCallback(
+      (globalErrors: string[] = []) => {
+        const deepHiddenFieldsErrors = hiddenFields?.map((hiddenField) => {
+          const hiddenFieldErrors = fieldErrors(hiddenField)
+
+          if (hiddenFieldErrors instanceof Array) {
+            const hiddenFieldLabel =
+              (labels && labels[hiddenField]) || inferLabel(String(hiddenField))
+            return hiddenFieldErrors.map(
+              (error) => `${hiddenFieldLabel}: ${error}`,
+            )
+          } else return []
+        })
+        const hiddenFieldsErrors: string[] =
+          deepHiddenFieldsErrors?.flat() || []
 
-      const allGlobalErrors = ([] as string[])
-        .concat(globalErrors, hiddenFieldsErrors)
-        .filter((error) => typeof error === 'string')
+        const allGlobalErrors = ([] as string[])
+          .concat(globalErrors, hiddenFieldsErrors)
+          .filter((error) => typeof error === 'string')
 
-      return allGlobalErrors.length > 0 ? allGlobalErrors : undefined
-    }
+        return allGlobalErrors.length > 0 ? allGlobalErrors : undefined
+      },
+      [fieldErrors, hiddenFields, labels],
+    )
 
-    let globalErrors = hiddenFieldsErrorsToGlobal(errors?._global)
+    const globalErrors = React.useMemo(
```

Comment:
Same comment here about the `useCallback`. Also, early caching might prevent us from doing proper profiling and optimizing the algorithms before implementing a caching strategy.